lib/RexStanSettings.php

```diff
@@ -0,0 +1,250 @@
+         * Sonderfall: developer-Addon.
+         */
+
+        $scanTargets = [];
+        foreach (rex_addon::getAvailableAddons() as $availableAddon) {
+            $scanTargets[$availableAddon->getPath()] = $availableAddon->getName();
+
+            if ('developer' === $availableAddon->getName() && class_exists(rex_developer_manager::class)) {
+                $scanTargets[rex_developer_manager::getBasePath() .'/modules/'] = 'developer: modules';
+                $scanTargets[rex_developer_manager::getBasePath() .'/templates/'] = 'developer: templates';
+            }
+        }
+
+        /**
+         * Die aktuelle PHP-Version des Servers (SAPI) markieren.
+         */
+
+        $version = (int) (PHP_VERSION_ID / 100) * 100;
+        $phpVersions = self::$phpVersionList;
+        if (isset($phpVersions[$version])) {
+            $phpVersions[$version] .= ' [aktuelle Webserver-Version (SAPI)]';
+        }
+
+        $field = $this->addInputField('number', 'level', null, ['class' => 'form-control', 'min' => 0, 'max' => 9]);
+        $field->setLabel('Level');
+        $field->setNotice('0 is the loosest and 9 is the strictest - <a href="https://phpstan.org/user-guide/rule-levels">see PHPStan Rule Levels</a>');
+
+        $field = $this->addSelectField('paths', null, ['class' => 'form-control selectpicker', 'data-live-search' => 'true', 'required' => 'required']); // die Klasse selectpicker aktiviert den Selectpicker von Bootstrap
+        $field->setAttribute('multiple', 'multiple');
+        $field->setLabel('AddOns');
+        $field->setNotice('AddOns, die untersucht werden sollen');
+        $select = $field->getSelect();
+        $select->addOptions($scanTargets);
+
+        $field = $this->addSelectField('includes', null, ['class' => 'form-control selectpicker']);
+        $field->setAttribute('multiple', 'multiple');
+        $field->setLabel('PHPStan Extensions');
+        $field->setNotice('Weiterlesen bzgl. der verf&uuml;gbaren Extensions: '.implode(', ', $extensionLinks));
+        $select = $field->getSelect();
+        $select->addOptions($extensions);
+
+        $field = $this->addSelectField('phpVersion', null, ['class' => 'form-control selectpicker']);
+        $field->setLabel('PHP-Version');
+        $field->setNotice('Referenz-Version für die Code-Analyse');
+        $select = $field->getSelect();
+        $select->addOptions($phpVersions);
+    }
+
+    /**
+     * Feldwerte abrufen aus userConf, defaultConf oder rex_config.
+     *
+     * Die Parent-Methode ruft ausschließlich Werte aus der Tabelle rex_config ab.
+     * Die überschriebene Methode versucht zunächst auf phpstan-Settings zuzugreifen.
+     * Kommt der NAme nicht in der Liste (switch..) vor, wird unter dem Namen auf
+     * rex_config zurückgegriffen.
+     *
+     * @param string $name  Der Feldname
+     *
+     * @return mixed  Der zum Feld gehörende Wert; default=''
+     */
+    protected function getValue($name)
```

Comment:
mir erschließt sich nicht warum das hier so kompliziert läuft. es war viel einfacher zuvor und ich sehe keinen problem was gegegnüber der vorherigen version gelöst wird

Reply:
In der vorigen Version wurde zwar ein rex_config-Formular benutzt, die Speicherung der phpstan-Werte aber komplett daneben abgearbeitet. Hier gibt's jetzt ein angepasstes Formular `class RexStanSettings extends rex_config_form`, dass dort, wo es vorgesehen ist, auch die Aktionen durchführt, nämlich in getValue zum Einlesen der Formularwerte und Save zum Speichern der Formularwerte..